### ISSUE TITLE
Proposal - Implement a way to tree shake constant icon font files (when there is dynamic use for others)

### ISSUE THREAD

**Issue by zambetpentru** [flutter/flutter#132034]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Use case

We have dynamic use of the Material Icon font so we can't tree shake that font file. It would be really useful to be able to tree shake the other fonts (e.g. Cupertino and Ionicons in our case) to minimize the Web bundle in particular.

### Proposal

For Flutter to be able to dynamically tree shake icon files that have const usage and to leave the other font files at their full scope.

**Comment by darshankawar** [comment]
@zambetpentru 
Can you take a look at [this](https://github.com/flutter/flutter/issues/64106) issue and check if your case resembles it or not ?

**Comment by zambetpentru** [comment]
Hi,

That looks to be a different good idea. I would like the existing tree shaking for fonts to work by excluding the non constant font, e.g. the material font icon, in my case.

**Comment by zambetpentru** [comment]
Related, as a manual workaround is there's way to manually create a small version of cupertinoicons file if I know the included packages only require a known list of icons?

**Comment by christopherfujino** [comment]
@zambetpentru please let me know if I understand your request:

You have multiple fonts:

1. Material icons, which has dynamic uses, and thus the static icon tree shaker would remove icons you might actually use at runtime
2. Cupertino and Ionicons, which use only const instances, and thus COULD be statically tree shaken.

And You would like to be able to run the icon tree shaker, but configure which fonts it operates on. Do I have this right?

**Comment by zambetpentru** [comment]
@christopherfujino Exactly! And in particular it won't build today without the --no-tree-shake-icons option, so that could instead tree shake the font files it can and earn about the excluded font files due to non const usage.

**Comment by christopherfujino** [comment]
> @christopherfujino Exactly! And in particular it won't build today without the --no-tree-shake-icons option, so that could instead tree shake the font files it can and warn about the excluded font files it couldn't due to non const usage.

Ahh yeah, I see. I could see changing from a boolean tree shake vs no tree shake, we could have a third mode where fonts that have only const references get tree-shaken, and those that have non-const references get a warning and no tree shaking.
